packages/swapper/src/swappers/zrx/buildQuoteTx/buildQuoteTx.ts

```diff
@@ -0,0 +1,191 @@
+import BigNumber from 'bignumber.js'
+import { AxiosResponse } from 'axios'
+import * as rax from 'retry-axios'
+import { ChainAdapter, ChainIdentifier } from '@shapeshiftoss/chain-adapters'
+import { Quote, SwapError, BuildQuoteTxArgs } from '../../..'
+import { ZrxSwapperDeps } from '../ZrxSwapper'
+import { applyAxiosRetry } from '../utils/applyAxiosRetry'
+import { erc20AllowanceAbi } from '../utils/abi/erc20-abi'
+import { normalizeAmount, getAllowanceRequired } from '../utils/helpers/helpers'
+import { zrxService } from '../utils/zrxService'
+import {
+  DEFAULT_SLIPPAGE,
+  DEFAULT_SOURCE,
+  DEFAULT_ETH_PATH,
+  AFFILIATE_ADDRESS,
+  APPROVAL_GAS_LIMIT
+} from '../utils/constants'
+
+type LiquiditySource = {
+  name: string
+  proportion: string
+}
+
+type QuoteResponse = {
+  price: string
+  guaranteedPrice: string
+  to: string
+  data?: string
+  value?: string
+  gas?: string
+  estimatedGas?: string
+  gasPrice?: string
+  protocolFee?: string
+  minimumProtocolFee?: string
+  buyTokenAddress?: string
+  sellTokenAddress?: string
+  buyAmount?: string
+  sellAmount?: string
+  allowanceTarget?: string
+  sources?: Array<LiquiditySource>
+}
+
+export async function buildQuoteTx(
+  { adapterManager, web3 }: ZrxSwapperDeps,
+  { input, wallet }: BuildQuoteTxArgs
+): Promise<Quote> {
+  const {
+    sellAsset,
+    buyAsset,
+    sellAmount,
+    buyAmount,
+    slippage,
+    sellAssetAccountId,
+    buyAssetAccountId,
+    priceImpact
+  } = input
+
+  if ((buyAmount && sellAmount) || (!buyAmount && !sellAmount)) {
+    throw new SwapError(
+      'ZrxSwapper:buildQuoteTx Exactly one of buyAmount or sellAmount is required'
+    )
+  }
+
+  if (!sellAssetAccountId || !buyAssetAccountId) {
+    throw new SwapError(
+      'ZrxSwapper:buildQuoteTx Both sellAssetAccountId and buyAssetAccountId are required'
+    )
+  }
+
+  const buyToken = buyAsset.tokenId || buyAsset.symbol || buyAsset.network
+  const sellToken = sellAsset.tokenId || sellAsset.symbol || sellAsset.network
+  if (!buyToken) {
+    throw new SwapError(
+      'ZrxSwapper:buildQuoteTx One of buyAssetContract or buyAssetSymbol or buyAssetNetwork are required'
+    )
+  }
+  if (!sellToken) {
+    throw new SwapError(
+      'ZrxSwapper:buildQuoteTx One of sellAssetContract or sellAssetSymbol or sellAssetNetwork are required'
+    )
+  }
+
+  // TODO: (ryankk) Remove the type cast when we unify ChainIdentifier and ChainTypes
+  const adapter: ChainAdapter = adapterManager.byChain(
+    (buyAsset.chain as unknown) as ChainIdentifier
+  )
+  const receiveAddress = await adapter.getAddress({ wallet, path: DEFAULT_ETH_PATH })
+
+  const slippagePercentage = slippage
```

Comment:
do we wanna put some sanity bounds around slippage here with an optional param to override sanity checks so people don't get rekt with 50% slippage?